File: submissions/sm_017_mukund/week_3/day_2/assignments/taxCalculator.js
var income=250000
var saving=00000
console.log("Income : ",income)
console.log("Saving : ",saving)
var Tax=0
var rebate=0
var TOI=0   // Final Tax on Income

if (income>1000000){
    rebate=(saving/10)
    if (rebate<=50000){
        TOI=income-rebate
        if (TOI>1000000){
            Tax=25000+100000+((3/10)*(TOI-1000000))
            console.log("Tax : ",Tax);
        }
        else{
            Tax=25000+((1/5)*(TOI-500000))
            console.log("Tax : ",Tax);
        }
    }
    else{
        TOI=income-50000
        if (TOI>1000000){
            Tax=25000+100000+((3/10)*(TOI-1000000))
            console.log("Tax : ",Tax);
        }
        else{
            Tax=25000+((1/5)*(TOI-500000))
            console.log("Tax : ",Tax);
        }
    }
}

else if ((500000<income) && (income<=1000000)){
    rebate=((3/10)*saving)
    if (rebate<=50000){
        TOI=income-rebate
        if (TOI>500000){
            Tax=25000+((1/5)*(TOI-500000))
            console.log("Tax : ",Tax);
        }
        else{
            Tax=((1/10)*(TOI-250000))
            console.log("Tax : ",Tax);
        }
    }
    else{
        TOI=income-50000
        if (TOI>500000){
            Tax=25000+((1/5)*(TOI-500000))
            console.log("Tax : ",Tax);
        }
        else{
            Tax=((1/10)*(TOI-250000))
            console.log("Tax : ",Tax);
        }
    }
}

else if ((250000<income) && (income<=500000)){
    rebate=(saving/2)
    if (rebate<=50000){
        TOI=income-rebate
        if (TOI>250000){
            Tax=((1/10)*(TOI-250000))
            console.log("Tax : ",Tax);
        }
        else{
            console.log("Tax : ",Tax);
        }
    }
    else{
        TOI=income-50000
        if (TOI>250000){
            Tax=((1/10)*(TOI-250000))
            console.log("Tax : ",Tax);
        }
        else{
            console.log("Tax : ",Tax);
        }
    }
}


else{
    console.log("Tax : ",Tax)
}

File: submissions/sm_028_sagar/week_13/day_5/session_1/tax_calculator.py
income = int(input())
saving = int(input())

#rebate calculation
rebate = 0

if(income > 250000 and income <= 500000):
    rebate = 0.5 * saving
elif (income > 500000 and income <= 1000000):
    rebate = 0.3 * saving
elif (income > 1000000):
    rebate = 0.1 * saving

if(rebate > 50000):rebate = 50000

actual_income = income - rebate

#tax-calculation
applicable_tax = 0

if(actual_income <= 250000):
    print('No Tax Applicable')
elif(actual_income > 250000 and actual_income <= 500000):
    applicable_tax = 0.1 * (actual_income - 250000)
elif(actual_income > 500000 and actual_income <= 1000000):
    applicable_tax = 0.2 * (actual_income - 500000) + 0.1 * 250000    
else:
    applicable_tax = 0.3 * (actual_income - 1000000) + 0.2 * 500000 + 0.1 * 250000

print('Applicable tax is ',applicable_tax)


#sample test case:
'''
500000
90000
('Applicable tax is ', 20500.0)
'''
'''
1000000
100000
('Applicable tax is ', 119000.0)
'''